src/dnsmasq/log.c

```diff
@@ -314,6 +314,19 @@ void my_syslog(int priority, const char *format, ...)
       fputc('\n', stderr);
     }
 
+  /* Pi-hole diagnosis system */
+  if(priority == LOG_WARNING)
```

Comment:
Where is `priority` set? Is this user-configurable? Is this a `dnsmasq` option?

Reply:
We only want to add `WARNING` level messages to the `message` table here. `dnsmasq` logs quite a few things like warnings, but also more generic information like queries. The priority is hard-coded in the `dnsmasq` code, some examples:

We want:

https://github.com/pi-hole/FTL/blob/a296265896876ba8178aac4394df51f0c1e9cb66/src/dnsmasq/dnsmasq.c#L955

https://github.com/pi-hole/FTL/blob/a296265896876ba8178aac4394df51f0c1e9cb66/src/dnsmasq/dnsmasq.c#L869

https://github.com/pi-hole/FTL/blob/a296265896876ba8178aac4394df51f0c1e9cb66/src/dnsmasq/dnsmasq.c#L1016

https://github.com/pi-hole/FTL/blob/198e7c61362e07b22baa8bdb2bb57dd1c53be0fc/src/dnsmasq/lease.c#L64-L66

We don't want:

https://github.com/pi-hole/FTL/blob/a296265896876ba8178aac4394df51f0c1e9cb66/src/dnsmasq/dnsmasq.c#L862

https://github.com/pi-hole/FTL/blob/a296265896876ba8178aac4394df51f0c1e9cb66/src/dnsmasq/dnsmasq.c#L1227

https://github.com/pi-hole/FTL/blob/2999e2b57c62b4455187ee9b77840d49df0a8e2e/src/dnsmasq/dump.c#L207

https://github.com/pi-hole/FTL/blob/db8ee8c6ac88c61c9ea160cce3fa4a84ca30ce5b/src/dnsmasq/blockdata.c#L55-L58